### PR TITLE
bpo-39199: Add attribution to the Green Tree Snakes in the AST docs

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue39199

### CODE REVIEW

File: Doc/library/ast.rst [hunk 1]
```diff
@@ -140,6 +140,11 @@ Node classes
    In the meantime, instantiating them will return an instance of
    a different class.
 
+.. note::
```
Comment (pablogsal) [inline Doc/library/ast.rst:143]: I have placed the note on the top, before the classes themselves as we already have a link to the project at the bottom of the page:

> See also Green Tree Snakes, an external documentation resource, has good details on working with Python ASTs.

**Comment by pablogsal** [comment]
CC @takluyver

File: Doc/library/ast.rst [hunk 2]
```diff
@@ -140,6 +140,11 @@ Node classes
...
+    The descriptions descriptions of the specific node classes displayed here
```
Comment (brandtbucher) [inline Doc/library/ast.rst]:
```suggestion
    The descriptions of the specific node classes displayed here
```

File: Doc/library/ast.rst [hunk 3]
```diff
@@ -140,6 +140,11 @@ Node classes
    In the meantime, instantiating them will return an instance of
    a different class.
 
+.. note::
+    The descriptions of the specific node classes displayed here
+    were initially adapted from the fantastic Green Tree Snakes `Green Tree
```
Comment (takluyver) [inline Doc/library/ast.rst]:
```suggestion
    were initially adapted from the fantastic `Green Tree
```

Putting the name once is enough :wink: 

Other than this, it looks good, thank-you!

Comment (pablogsal) [reply Doc/library/ast.rst]: Haha, ResT links will be the end of me 😅

**Review by willingc — APPROVED** [review]

**Comment by willingc** [comment]
Thanks @pablogsal. As soon as @takluyver approves of the content, let's ship this.

**Comment by pablogsal** [comment]
> Thanks @pablogsal. As soon as @takluyver approves of the content, let's ship this. One of these days I would love to see @takluyver have core dev privileges.

Thanks for the review. @takluyver approved in this comment:

https://github.com/python/cpython/pull/24727#discussion_r586623451

So I am landing this :)

**Pull request merged**